src/index.js

```diff
@@ -109,6 +109,11 @@ class AEMHeadless {
       return `;${key}=${val}`
     }).join(''))
 
+    if (method === 'GET') {
+      // https://experienceleague.adobe.com/en/docs/experience-manager-cloud-service/content/headless/graphql-api/content-fragments#dynamic-image-delivery-multiple-specified-parameters
+      variablesString += variablesString + ';';
```

Comment:
@rbartl-pv done - let me know if there is anything else to address. :)  
Unfortunately I couldn't verify the queries or tests with two parameters as there is no local model or test data model I could run queries against with two parameters except the mocks.

Reply:
@colinscz : Sorry, I was not attentive enough at my first look
Instead of
`variablesString += variablesString + ';';`
it should be
`variablesString += ';'`

Also the expected URL in the test 'API: runPersistedQuery GET with variables API Success' would have to be updated to (the additional trailing semi-colon)
```
expect(fetch).toHaveBeenCalledWith(`http://localhost/graphql/execute.json/${persistedName}%3Bname%3Dtest;`, expect.anything(), expect.anything())
```

And in the test you added it should be:
```
expect(fetch).toHaveBeenCalledWith(`http://localhost/graphql/execute.json/${persistedName}%3Bname%3Dtest%3Blocale%3Den;`, expect.anything(), expect.anything())
```

That's for the validation to pass. You can do this locally by running:
```
npm install
npm test
```

Since the PR comes from a fork please do the adaptions there. After that I will still need to check if we're good with that because we will have the semicolons partially encoded except the last one which is not.
Theoretically you should be able to test this also against a real AEM instance if available.

Reply:
@rbartl-pv found two more issues in the test and the implementation that I fixed. The build and tests are now running through

Reply:
f7d4a6864da179c6488a69d842473a21023ae649

Reply:
@colinscz 

The checks passed now and the PR is approved: but before merging and releasing a new version I'd like to do some testing against a local AEM instance in the next 2 days (probably tomorrow).

Reply:
@rbartl-pv ok sounds good - we are making use of the node-js package, which depends on this one, so probably a release of that one will need to happen in order to help both SPA and SSR use cases :)

Reply:
@colinscz : We still need a little bit more of time since our team has recently took the maintenance/ownership of this repo and we need to still align with the previous contributors regarding the release process, etc. - so I hope that we will merge this PR/make a new release towards the mid/end of next week. Thanks for your understanding!

Reply:
@colinscz : I've run now the e2e tests in https://github.com/adobe/aem-headless-client-js/tree/main/e2e
and I noticed another problem. We should append the trailing semicolon only if variablesString is not empty, otherwise queries without variables will fail.
So it should be something like:
`if (method === 'GET' && variablesString) {`

If you could also add an additional unit-test-case for this would be great.
And if you even add more e2e tests to cover also variables (to execute them you'll need a local AEM with the wknd-sample-application installed).
Otherwise it's also ok - I could add that also later on.

Reply:
@rbartl-pv good point - I've done the changes requested regarding the unit test and the empty variableString.

However it's not so easy for me to get a local AEM instance running, so it would be much quicker if you could add those e2e testing scenarios and test them on your side.
For the future it would greatly help external contributors especially with e2e tests if there is some easy way to run or connect to a dedicated sandbox or test instance of AEM in the future or if this can be started up and is part of the repository.

Reply:
@rbartl-pv ok thanks a lot! If you get the chance to do a release of the depending nodejs server package https://github.com/adobe/aem-headless-client-nodejs with this fix it would be appreciated.
If it helps I can prepare a PR for that repository as well.

> I am afraid that this is not so easy possible.

Hmm unfortunate but understandable since the setup doesn't seem to be so quick and easy.

Reply:
https://github.com/adobe/aem-headless-client-nodejs/pull/17